source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -19,6 +19,11 @@ AMDSClientTCPSocket::AMDSClientTCPSocket(const QString host, const quint16 port,
 	socketKey_ = "";
 	tcpSocket_ = new QTcpSocket(this);
 
+	waitingMorePackages_ = false;
+	readedBufferSize_ = 0;
+	expectedBufferSize_ = 0;
+	incomeDataBuffer_ = new QByteArray();
```

Comment:
You can do that, or have a cleanup method, or delete the buffer in the destructor.  Any of these are valid options because you don't lose anything in performance if you pass `const QByteArray &`'s around because of Qt's implicit sharing.